### PR TITLE
Fix(snackbar): add the same format for future implementation

### LINKED ISSUE #690: Snackbar appears after close
After singing up, appears a snackbar showing a successful message, after a close, it keeps appearing.

Steps to test:
1. Sign up
2. Close successful snackbar
3. Check it appears again

image

### PR DESCRIPTION
**Pull request by Edgar097**
Resolves #690 #636 #526

### What does this PR do?
Add the same implementation for all the snackbars in the app

### How should this be tested?
- Make run
- Navigate through the app
- Check if the snackbars work properly

### CODE REVIEW

File: webapp/src/components/CredentialsRecovery/CredentialsRecovery.js [hunk 1]
```diff
@@ -99,7 +94,11 @@ const CredentialsRecovery = ({ overrideBoxClass, overrideLabelClass }) => {
         }
       })
       setValidEmailFormat(false)
-    } else setErrorMessage(t('credentialsRecovery.passwordNotChangable'))
+    } else setOpenSnackbar({
+      show: true,
+      message: t('credentialsRecovery.passwordNotChangable'),
```
Comment (JustinCast) [inline webapp/src/components/CredentialsRecovery/CredentialsRecovery.js:99]: ~passwordNotChangable~ passwordNotChangeable?

File: webapp/src/components/CredentialsRecovery/CredentialsRecovery.js [hunk 2]
```diff
@@ -82,12 +74,15 @@ const CredentialsRecovery = ({ overrideBoxClass, overrideLabelClass }) => {
...
```
Comment (JustinCast) [inline webapp/src/components/CredentialsRecovery/CredentialsRecovery.js:79]: ~passwordNotChangable~ passwordNotChangeable?

File: webapp/src/components/DonationsDashboard/DonationsDashboard.js [hunk 3]
```diff
@@ -112,25 +110,31 @@ const DonationsDashboard = ({ isDesktop, currentUser, isOffer }) => {
...
 
-  const handleOpenAlert = () => setOpenAlert(!openAlert)
-
-  const handleSnackbarClose =  () => {
-    setSuccess(false)
-    setErrorMessage(null)
+  const handleSnackbarClose = () => {
+    setOpenSnackbar({ ...openSnackbar, show: false })
     setAccountTo()
+
```
Comment (JustinCast) [inline webapp/src/components/DonationsDashboard/DonationsDashboard.js:116]: Delete this blank space

File: webapp/src/components/DonationsDashboard/DonationsDashboard.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import React, { useState, useEffect, forwardRef } from 'react'
-import { useMutation, useLazyQuery , useSubscription } from '@apollo/react-hooks'
+import React, { useState, useEffect, forwardRef, useCallback } from 'react'
```
Comment (JustinCast) [inline webapp/src/components/DonationsDashboard/DonationsDashboard.js:1]: Are you using useCallback hook?

File: webapp/src/components/LoginModal/Login.js [hunk 5]
```diff
@@ -238,24 +251,23 @@ const LoginModal = ({ isNavBar, isSideBar }) => {
...
-              className={classes.alert}
-              severity="error"
+              severity={openSnackbar.severity}
               action={
                 <IconButton
                   aria-label="close"
                   color="inherit"
                   size="small"
-                  onClick={() => setErrorMessage(null)}
+                  onClick={() => handleOpenAlert()}
```
Comment (JustinCast) [inline webapp/src/components/LoginModal/Login.js:262]: onClick={handleOpenAlert}

**Review by JustinCast — CHANGES_REQUESTED** [review]

**Comment by JustinCast** [comment]
@Edgar097 take a look at Codefactor's warnings!

**Review by JustinCast — CHANGES_REQUESTED** [review]
@Edgar097 take a look at this 'setErrorMessage' is not defined. (no-undef) codefactor warning

**Comment by JustinCast** [comment]
@Edgar097 also take a look at the conflicts

**Review by JustinCast — APPROVED** [review]

**Review by leisterfrancisco — APPROVED** [review]

**Pull request merged**